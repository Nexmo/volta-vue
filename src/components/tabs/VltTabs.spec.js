import { mount } from '@vue/test-utils';
import { h } from 'vue';
import { VltTabs, VltTab } from '..';

const tabClass = 'Vlt-tabs__link';
const activeTabClass = `Vlt-tabs__link_active ${tabClass}`;

describe('vlt-tabs', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount({
      render() {
        return h(VltTabs, {}, [
          h(VltTab, { label: 'One' }),
          h(VltTab, { label: 'Two' }),
          h(VltTab, { label: 'Three' }),
        ]);
      },
    });
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('second tab can be selected', async () => {
    const secondTab = wrapper.findAll('li').at(1);
    secondTab.trigger('click');
    await wrapper.vm.$nextTick();
    expect(secondTab.attributes('class')).toBe(activeTabClass);
  });

  test('second tab can be selected, then third tab', async () => {
    const secondTab = wrapper.findAll('li').at(1);
    secondTab.trigger('click');
    await wrapper.vm.$nextTick();
    expect(secondTab.attributes('class')).toBe(activeTabClass);

    const thirdTab = wrapper.findAll('li').at(2);
    thirdTab.trigger('click');
    await wrapper.vm.$nextTick();
    expect(secondTab.attributes('class')).toBe(tabClass);
    expect(thirdTab.attributes('class')).toBe(activeTabClass);
  });
});
