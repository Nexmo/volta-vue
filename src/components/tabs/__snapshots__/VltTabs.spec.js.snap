// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-tabs renders correctly 1`] = `
<div
  class="Vlt-tabs"
>
  <ul
    class="Vlt-tabs__header"
  >
    
    <li
      class="Vlt-tabs__link_active Vlt-tabs__link"
      id=""
    >
      <!--v-if-->
      <span>
        One
      </span>
    </li>
    <li
      class="Vlt-tabs__link"
      id=""
    >
      <!--v-if-->
      <span>
        Two
      </span>
    </li>
    <li
      class="Vlt-tabs__link"
      id=""
    >
      <!--v-if-->
      <span>
        Three
      </span>
    </li>
    
  </ul>
  <div
    class="Vlt-tabs__content"
  >
    
    <div
      class="Vlt-tabs__panel Vlt-tabs__panel_active"
    >
      
      
    </div>
    <div
      class="Vlt-tabs__panel"
    >
      
      
    </div>
    <div
      class="Vlt-tabs__panel"
    >
      
      
    </div>
    
  </div>
</div>
`;
